projects/valtimo/components/src/lib/services/cds-theme.service.ts

```diff
@@ -46,31 +53,53 @@ export class CdsThemeService implements OnDestroy {
   }
 
   private openDarkModeSubscription(): void {
-    combineLatest([this.getBrowserPrefersDarkModeObservable(), this.preferredTheme$]).subscribe(
-      ([browserPrefersDarkMode, preferredTheme]) => {
-        switch (preferredTheme) {
-          case SelectableCarbonTheme.SYSTEM:
-            if (browserPrefersDarkMode) {
-              this.setThemeOnDocument(SelectableCarbonTheme.G90);
-            } else {
-              this.setThemeOnDocument(SelectableCarbonTheme.G10);
-            }
-            break;
-          case SelectableCarbonTheme.G10:
-            this.setThemeOnDocument(SelectableCarbonTheme.G10);
-            break;
-          case SelectableCarbonTheme.G90:
-            this.setThemeOnDocument(SelectableCarbonTheme.G90);
-            break;
-        }
-      }
+    this._subscriptions.add(
+      this.preferredTheme$
+        .pipe(
+          switchMap((preferredTheme: SelectableCarbonTheme) =>
```

Comment:
Why this refactor? It seems a lot less readable to me

Reply:
Combination of adding functionality for Monaco editor, and removing dual initialisation of themes causing the Monaco editor theming to break.